packages/react-components/src/components/table-output-component.tsx

```diff
@@ -252,12 +272,16 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
     }
 
     private onModelUpdated = async () => {
-        if (this.pendingIndex && this.gridApi) {
-            if (this.gridApi.getSelectedNodes().length === 0) {
-                this.gridApi.getDisplayedRowAtIndex(this.pendingIndex).setSelected(true);
+        this.pendingIndices.forEach(index => {
+            if (this.gridApi) {
+                const node = this.gridApi.getDisplayedRowAtIndex(index);
+                if (node.id) {
+                    node.setSelected(true);
+                    this.gridApi.ensureIndexVisible(index);
+                }
             }
-            this.pendingIndex = undefined;
-        }
+        });
+        this.pendingIndices = [];
```

Comment:
Is it sure that all pending indices have been loaded? Otherwise you need to only remove those that were found.